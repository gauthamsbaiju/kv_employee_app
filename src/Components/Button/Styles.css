.createbtn{
    background: #03AEEE 0% 0% no-repeat padding-box;
    box-shadow: 0px 3px 6px #BFEEFF;
    border-radius: 3px;
    border: 0px;
    color: white;
    height: 100%;
    
    width: 100%;
}

.cancelbtn{
    height: 100%;
    width: 100%;
    background: #FFFFFF 0% 0% no-repeat padding-box;
    box-shadow: 0px 3px 6px #C9C9C929;
    border: 1px solid #8A8C8F;
    border-radius: 3px;
    /* width: calc(100% *   0.2); */
}

.button-wrapper{
    width: 45%;
    height: 35px;
}