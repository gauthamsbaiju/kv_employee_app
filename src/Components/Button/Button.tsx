import React from 'react';
import './Styles.css';

type InputPropsType = {
  label: string;
  className: string;
  onClick: (e) => void;
  type: 'submit';
};

const Button: React.FC<InputPropsType> = (props) => {
  return (
    <div className='button-wrapper'>
      <button className={props.className} onClick={props.onClick} type={props.type}>
        {props.label}
      </button>
    </div>
  );
};

export default Button;
