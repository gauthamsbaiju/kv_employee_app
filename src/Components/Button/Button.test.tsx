import React from 'react';
import Button, { ButtonPropsType } from './Button';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('If Button works properly', () => {
  test('If label rendered correctly', () => {
    const ButtonProps: ButtonPropsType = {
      label: 'Button',
      type: 'submit'
    };

    render(<Button {...ButtonProps} />);
    const element = screen.getByTestId('button-test');

    if (element) expect(element).toHaveTextContent('Button');
  });
  test('If snapshot rendered', () => {
    const ButtonProps: ButtonPropsType = {
      label: 'Button',
      type: 'submit'
    };

    render(<Button {...ButtonProps} />);
    const element = screen.getByTestId('button-test');

    expect(element).toMatchSnapshot();
  });
  test('If type is correct', () => {
    const ButtonProps: ButtonPropsType = {
      label: 'Button',
      type: 'submit'
    };

    render(<Button {...ButtonProps} />);
    const element = screen.getByTestId('button-test');

    if (element) expect(element).toHaveAttribute('type', 'submit');
  });

  test('If onClick is triggered', () => {
    const onClick = jest.fn();
    const ButtonProps: ButtonPropsType = {
      label: 'Button',
      type: 'submit',
      onClick
    };

    render(<Button {...ButtonProps} />);
    const element = screen.getByTestId('button-test');

    fireEvent.click(element);
    expect(onClick).toBeCalled();
  });
});
