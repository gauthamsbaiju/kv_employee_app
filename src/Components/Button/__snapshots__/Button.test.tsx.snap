// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If Button works properly If snapshot rendered 1`] = `
<button
  data-testid="button-test"
  type="submit"
>
  Button
</button>
`;
